Type hints, code formatting, utility functions

liquidz00 opened the pull request:
## Added
- `ruff.toml` file for consistent code formatting while respecting existing preferences (single-quotes instead of double quotes)
- (trivial) `load_json` function for reuse in `jawa_receiver.py` module

## Changed
- Type hints for function parameters and return values

## Removed
- Unused variables and exception strings ([example](https://github.com/jamf/JAWA/blob/bc545e6e110e6a16d540e99bebd3cb7940bb79bb/views/home_view.py#L201-L204))

If exception string was relevant to workflow, added f-string with exception message into logs/return statements/etc.

ball42 reviewed (changes requested):
Type hints and util functions look good, but can we adjust the ruff.toml quote-style setting to "double" instead?

liquidz00 commented:
> Type hints and util functions look good, but can we adjust the ruff.toml quote-style setting to "double" instead?

Definitely! Not entirely sure what the behavior would have been if I removed the setting entirely, so instead I configured `ruff.toml` to use double quotes. The rest of the codebase has been formatted accordingly. 

Lastly, I added the `.ruff cache` directory to `.gitignore`. Mistakenly forgot to include that in the original PR, my bad!
